src/main/java/com/iexec/blockchain/tool/IexecHubService.java

```diff
@@ -115,6 +119,23 @@ public TransactionReceipt finalizeTask(String chainTaskId,
                 .send();
     }
 
+    /**
+     * Synchronized sleep to ensure several transactions will never be sent in the same time interval.
+     * <p>
+     * This synchronized sleep is required for nonce computation on pending block.
+     * When a first transaction will be emitted, it will be emitted and registered on the pending block.
+     * After a latency, a second transaction can be sent on the pending block and the nonce will be computed successfully.
+     * With a correct nonce, it becomes possible to perform several transactions from the same wallet in the same block.
+     *
+     * @throws InterruptedException if the calling thread is interrupted
+     */
+    private synchronized void addLatency() throws InterruptedException {
+        final long deadline = Instant.now().plus(1L, ChronoUnit.SECONDS).toEpochMilli();
+        while (Instant.now().toEpochMilli() < deadline) {
+            wait(deadline - Instant.now().toEpochMilli());
+        }
```

Comment:
Is it better than `TimeUnit.SECOND.wait(1)`? Why do you need 3 lines to wait for 1 second? :thinking:

Reply:
If we don't do this, sonar analysis tools will easily trigger the following rules:
* `Thread.sleep` usage : https://rules.sonarsource.com/java/RSPEC-2276/
* `TimeUnit.wait` or `wait` without protecting it with a loop and a condition:
   https://rules.sonarsource.com/java/type/Code%20Smell/RSPEC-2274/

Reply:
Thanks for the analysis and links